receiver/sqlqueryreceiver/metrics.go

```diff
@@ -46,6 +50,22 @@ func rowToMetric(row metricRow, cfg MetricCfg, dest pmetric.Metric) error {
 	return nil
 }
 
+func setTimestamp(cfg MetricCfg, dp pmetric.NumberDataPoint, ts pcommon.Timestamp, scrapeCfg scraperhelper.ScraperControllerSettings) {
+	// default is gauge but add ts to all metrics
+	// start time is not needed on gauge metrics
+	dp.SetTimestamp(ts)
+
+	// Cumulative sum should have a start time set to the beginning of the data points cumulation
+	if cfg.Aggregation == MetricAggregationCumulative && cfg.DataType != MetricDataTypeGauge {
+		dp.SetStartTimestamp(ts)
```

Comment:
Oh I missed one more thing. This should be receiver start time, not the current timestamp. Save the receiver start time during in the scraper Start method and use it here.